### PR TITLE
[9.x] Collection select()

### PR DESCRIPTION
**Pull request by roberto-butti**
As explained here: https://github.com/laravel/framework/discussions/41521 , I open a PR for introducing _select()_ method for Collection.
_select()_ get data from bi-dimensional array by keys.

I created also tests in tests/Support/SupportCollectionTest.php adding _testSelect()_ method with some test case (not existent columns, null columns, string column.

### CODE REVIEW

File: src/Illuminate/Collections/Arr.php [hunk 1]
```diff
@@ -597,6 +597,25 @@ public static function random($array, $number = null, $preserveKeys = false)
...
+    /**
+     * Select and "pick" columns from a bi-dimensional array.
+     *
+     * @param  array  $array
+     * @param  array  $columns
+     * @return array
+     */
+    public static function select(&$array, $columns)
+    {
+        if (is_string($columns)) {
```
Comment (hAz5) [inline src/Illuminate/Collections/Arr.php:609]: I think You can use 
```
$columns = (array) $columns;
```

instead of two If conditions.

Comment (roberto-butti) [reply src/Illuminate/Collections/Arr.php:609]: > I think You can use
> 
> ```
> $columns = (array) $columns;
> ```
> 
> instead of two If conditions.

Thank you for the feedback.
Yes i think that your suggestion is great.
```
>>> (array) "some"
=> [
     "some",
   ]
>>> (array) null
=> []

```

File: src/Illuminate/Collections/Arr.php [hunk 2]
```diff
@@ -597,6 +597,20 @@ public static function random($array, $number = null, $preserveKeys = false)
         return $results;
     }
 
+    /**
+     * Select and "pick" columns from a bi-dimensional array.
+     *
+     * @param  array  $array
+     * @param  array  $columns
```
Comment (dennisprudlo) [inline src/Illuminate/Collections/Arr.php:604]: Shouldn't this be `@param  array|string  $columns` then, if we accept strings as well?

File: src/Illuminate/Collections/Collection.php [hunk 3]
```diff
@@ -1046,6 +1046,17 @@ public function search($value, $strict = false)
         return false;
     }
 
+    /**
+     * Select and "pick" columns from a bi-dimensional array.
+     *
+     * @param  array  $columns
```
Comment (dennisprudlo) [inline src/Illuminate/Collections/Collection.php:1052]: Same thing here

**Comment by roberto-butti** [comment]
> 
> If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions!
> 

ok , thank you for the suggestion.
I'm building a package for this

```php
use HiFolks\DataType\Table;

$table = Table::make([
    ['product' => 'Desk', 'price' => 200, 'active' => true],
    ['product' => 'Chair', 'price' => 100, 'active' => true],
    ['product' => 'Door', 'price' => 300, 'active' => false],
    ['product' => 'Bookcase', 'price' => 150, 'active' => true],
    ['product' => 'Door', 'price' => 100, 'active' => true],
]);
/**
 * I would like to **filter** the rows with price greater than 100,
 * select only "product" and "price" fields,
 * and for each rows **create a new field** named "new_filed"
 * that is a calculated field (doubling the price)
 */
var_export(
    $table
        ->select(['product' , 'price'])
        ->where('price', 100, '>')
        ->calc("new_field", fn ($item) => $item['price'] * 2)
        ->arr()
);
/*
array (
  0 => array (
    'product' => 'Desk',
    'price' => 200,
    'new_field' => 400,
  ),
  2 => array (
    'product' => 'Door',
    'price' => 300,
    'new_field' => 600,
  ),
  3 => array (
    'product' => 'Bookcase',
    'price' => 150,
    'new_field' => 300,
  ),
)
 */
```

The packages is https://github.com/Hi-Folks/array , the class is **Table**
